mobile/lib/services/location_service.dart

```diff
@@ -54,6 +54,16 @@ class LocationService {
   }
 
   static Future<CurrentLocation?> getCurrentLocation() async {
+    final locationGranted = await LocationService.locationGranted();
+    if (!locationGranted) {
+      return null;
+    }
+
+    final bool serviceEnabled = await Geolocator.isLocationServiceEnabled();
+    if (!serviceEnabled) {
+      return null;
+    }
```

Comment:
checking whether location services are available before attempting to get a user's location.